packages/middleware-signing/src/awsAuthConfiguration.ts

```diff
@@ -196,12 +204,18 @@ export const resolveAwsAuthConfig = <T>(
 
       const signingRegion = authScheme.signingRegion;
       const signingService = authScheme.signingName;
-      // update client's singing region and signing service config if they are resolved.
-      // signing region resolving order: user supplied signingRegion -> endpoints.json inferred region -> client region
-      input.signingRegion = input.signingRegion || signingRegion;
-      // signing name resolving order:
-      // user supplied signingName -> endpoints.json inferred (credential scope -> model arnNamespace) -> model service id
-      input.signingName = input.signingName || signingService || input.serviceId;
+
+      if (overwrite) {
```

Comment:
nit: `overwrite` reads like the `input.*` parameters would be overwritten, but the code path here is using the original `input.*` parameters.

Reply:
unfortunately this part of the code mutates the config during signing and not during the constructor config resolution

Reply:
I changed it to only mutate if authscheme is not sigv4a. Sigv4a doesn't contain a meaningful region so far, so we can skip writing that to the input and leave it to any further resolutions containing a singular region.